File: cscs-checks/prgenv/src/openmp/hello_world_openmp.cpp
#include <stdio.h>   
#include <omp.h>
 
int main(int argc, char *argv[])
{
  int tid, nthreads;
  #pragma omp parallel private(tid, nthreads)
  {
    tid = omp_get_thread_num();
    nthreads = omp_get_num_threads();
    #pragma omp critical
    {
      printf("Hello World from thread %d out of %d from process %d out of %d\n",
       tid, nthreads, 0, 1);
    }
  }
 
  return 0;
}

File: cscs-checks/prgenv/src/openmp/hello_world_openmp.f90
program hello90
use omp_lib
integer:: id, nthreads
  !$omp parallel private(id, nthreads)
  id = omp_get_thread_num()
  nthreads = omp_get_num_threads()
  write (*,'(A23,1X,I3,1X,A6,1X,I3,1X,A12,1X,I3,1X,A6,I3)') 'Hello World from thread', id, &
	'out of', nthreads, 'from process', 0, 'out of', 1

  !$omp end parallel
end program

